src/Core/SharedKernels/Enumerator/Enumerator/Implementation/Services/InterviewerInterviewAccessor.cs

```diff
@@ -139,28 +141,46 @@ private FeaturedQuestionMeta ToFeaturedQuestionMeta(PrefilledQuestionView prefil
                 prefilledQuestion.Answer);
         }
 
-        private IReadOnlyCollection<CommittedEvent> GetAllEventsToSend(Guid interviewId)
+        private IReadOnlyCollection<CommittedEvent> GetAllEventsToSend(Guid interviewId,
+            SyncInfoPackageResponse syncInfo)
         {
-            var minVersion = this.eventStore.GetLastEventKnownToHq(interviewId) + 1;
+            var minVersion = syncInfo.NeedSendFullStream || !syncInfo.HasInterview
+                ? 0
+                : this.eventStore.GetLastEventKnownToHq(interviewId) + 1;
             return this.eventStore.Read(interviewId, minVersion)
                 .ToReadOnlyCollection();
         }
 
-        private IReadOnlyCollection<CommittedEvent> GetFilteredEventsToSend(Guid interviewId)
+        private IReadOnlyCollection<CommittedEvent> GetFilteredEventsToSend(Guid interviewId,
+            SyncInfoPackageResponse syncInfo)
         {
             var lastCompleteSequence = this.eventStore.GetMaxSequenceForAnyEvent(interviewId, nameof(InterviewCompleted), nameof(InterviewModeChanged));
             var lastComplete = this.eventStore.GetEventByEventSequence(interviewId, lastCompleteSequence);
 
+            var minVersion = syncInfo.NeedSendFullStream || !syncInfo.HasInterview
+                ? 0
+                : this.eventStore.GetLastEventKnownToHq(interviewId) + 1;
+
             return this.eventStreamOptimizer.FilterEventsToBeSent(
-                this.eventStore.Read(interviewId, this.eventStore.GetLastEventKnownToHq(interviewId) + 1),
+                this.eventStore.Read(interviewId, minVersion),
                 lastComplete?.CommitId);
         }
 
-        public InterviewPackageContainer GetInterviewEventStreamContainer(Guid interviewId, bool needCompress)
+        public InterviewSyncInfoPackage GetInterviewSyncInfoPackage(Guid interviewId)
+        {
+            return new InterviewSyncInfoPackage()
+            {
+                FirstEventId = this.eventStore.GetFirstEventId(interviewId)!.Value,

```

Comment:
Could you please instead of suppressing NRE check add either explicit check and throw or update method GetFirstEventId to GetFirstEventIdOrThrow

Reply:
Added check on empty event stream